concrete/src/Notification/Alert/Filter/FilterListFactory.php

```diff
@@ -0,0 +1,39 @@
+<?php
+namespace Concrete\Core\Notification\Alert\Filter;
+
+use Concrete\Core\Notification\Type\Manager;
+use Concrete\Core\Notification\Type\TypeInterface;
+
+class FilterListFactory
+{
+
+    /**
+     * @var Manager
+     */
+    protected $typeManager;
+
+    public function __construct(Manager $typeManager)
+    {
+        $this->typeManager = $typeManager;
+    }
+
+    /**
+     * @return FilterList
+     */
+    public function createList()
+    {
+        $list = new FilterList();
+        foreach($this->typeManager->getDrivers() as $driver) {
+            /**
+             * @var $driver TypeInterface
+             */
+            foreach($driver->getAvailableFilters() as $filter) {
```

Comment:
If this method had a return type set to:
```
/**
 * @return TypeInterface[]
 */
```
you wouldn't need to declare this `@var`

Reply:
Unfortunately, getDrivers() is something on one of our low level manager classes and it doesn't know what it's returning